components/alert/test/alert-toast.test.js

```diff
@@ -12,6 +12,17 @@ describe('d2l-alert-toast', () => {
 
 	});
 
+	describe('button-press', () => {
+
+		it('should fire "d2l-alert-toast-button-press" event when alert button is pressed', async() => {
+			const el = await fixture(html`<d2l-alert-toast button-text="Click Me" open>message</d2l-alert-toast>`);
+			const alert = el.shadowRoot.querySelector('d2l-alert');
+			setTimeout(() => alert.dispatchEvent(new CustomEvent('d2l-alert-button-press', { bubbles: true, composed: true })));
```

Comment:
This test assumes that `d2l-alert` continues to dispatch `d2l-alert-button-press`. If that changed, this test would still pass but the toast component would no longer be working correctly. A more comprehensive test would call `click()` on the underlying button, but it requires poking even further into the component's local DOM.

Reply:
Yeah, both ways are subject to being broken if the component is changed in the future (e.g. if the button is changed). My train of thought was that this test should not be testing the inner workings of `<d2l-alert>`, rather it should assume that `<d2l-alert>` works and use the API to test the `<d2l-alert-toast>` component. But you're right that if the `<d2l-alert>` API is changed, then it would not break this test.

Reply:
Yep that's fair.